### PR TITLE
Named fragment support.

### PR DESCRIPTION
**Pull request by Poincare**
Named fragment support within the client. Now passes around the fragments within a query as a symbol table (i.e. an object mapping from fragment name to the fragment definition). 

TODO:

- [x] Update CHANGELOG.md with your change
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

### CODE REVIEW

**Comment by Poincare** [comment]
@helfer - could you review?

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Expect active development and potentially significant breaking changes in the `0
 
 ### vNEXT
 - Removed AuthTokenHeaderMiddleware code and related tests from apollo-client [Issue #247](https://github.com/apollostack/apollo-client/issues/247)
-- Fixed the getFromAST functions so that they work with queries that have named fragments (i.e. working toward named fragment support)
```
Comment (stubailo) [inline CHANGELOG.md]: This needs an issue and PR link please!

File: src/QueryManager.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ import {
   getMutationDefinition,
   getQueryDefinition,
   replaceOperationDefinition,
+  getFragmentDefinitions,
+  createFragmentSymTable,
```
Comment (stubailo) [inline src/QueryManager.ts]: I feel like "symtable" will be odd for people who are not familiar with that concept - perhaps we can use a simpler name like "fragmentMap"?

**Comment by Poincare** [comment]
Made the changes.

File: test/getFromAST.ts [hunk 3]
```diff
@@ -200,4 +202,22 @@ describe('AST utility functions', () => {
     const newDoc = replaceOperationDefinition(queryWithFragments, newQueryDef);
     assert.equal(print(newDoc), print(expectedNewQuery));
   });
+
+  it('should create the fragment sym table correctly', () => {
```
Comment (helfer) [inline test/getFromAST.ts]: should be renamed to fragment map as well

**Comment by helfer** [comment]
Just one tiny comment about one of the tests. Ready to merge otherwise. 🎉

**Comment by Poincare** [comment]
Fixed.

**Comment by stubailo** [comment]
Looks awesome!

**Comment by helfer** [comment]
Hm, we seem to have merged something in the meantime. Can you rebase one last time?

**Comment by Poincare** [comment]
Rebased.

**Comment by helfer** [comment]
Merged!

**Pull request merged**